tsdb/chunkenc/chunk.go

```diff
@@ -52,6 +52,24 @@ func IsValidEncoding(e Encoding) bool {
 	return e == EncXOR || e == EncHistogram || e == EncFloatHistogram
 }
 
+const (
+	// MaxBytesPerXORChunk is a soft cap for the size of an XOR chunk. It is a soft cap in that a final
+	// sample may be appended that exceeds the cap but no further samples will be appended to that chunk.
+	// This results in an effective hard cap of 1024 bytes (1005 bytes already in a chunk followed by a
+	// sample that takes the maximum size of 19 bytes).
+	MaxBytesPerXORChunk = 1005
+	// TargetBytesPerNativeHistogramChunk sets a size target for each native histogram chunk.
+	TargetBytesPerNativeHistogramChunk = 1024
+	// MinSamplesPerNativeHistogramChunk sets a minimum sample count for native histogram chunks. This is
+	// desirable because a single native histogram sample can be larger than TargetBytesPerNativeHistogramChunk
+	// but we want to avoid too-small sample count chunks so we can achieve some measure of compression advantage
+	// even while dealing with really large histograms.
+	// Note that this minimum sample count is not enforced across chunk range boundaries (for example, if the chunk
+	// range is 100 and the first sample in the chunk range is 99, the next sample will be included in a new chunk
+	// resulting in the old chunk containing only a single sample).
+	MinSamplesPerNativeHistogramChunk = 10
```

Comment:
@beorn7 do you think we can reduce it by any chance?

Reply:
This is just a ballpark number (as reasoned earlier, min amount of ten samples gives us 90% of the max possible compression, and we humans like 9s… – we could as well set this to any other number. Much larger won't help a lot, as we have already utilized 90% of the max compression, but we also shouldn't go much lower to not kill the compression entirely. If this number is 8 or 7 or what not doesn't make a huge dent, but every step we approach 1 is making a progressively larger dent… the good thing is that we can play with this without breaking anything).